File: ios/YaMap/YaMapVC.m
//
//  YaMapVC.m
//  YaMapTest
//
//  Created by Pavel on 16.08.2023.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

#import <React/RCTViewManager.h>
 
@interface RCT_EXTERN_MODULE(YaMapVC, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(pointsJson, NSString)
RCT_EXPORT_VIEW_PROPERTY(styleJson, NSString)
RCT_EXPORT_VIEW_PROPERTY(zoom, NSNumber)
RCT_EXPORT_VIEW_PROPERTY(onPressMarker, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onMapPressed, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onCameraPositionChangedEnd, RCTBubblingEventBlock)
RCT_EXTERN_METHOD(setCenter:(nonnull NSNumber *)reactTag center:(NSDictionary *_Nonnull)center zoom:(NSNumber *_Nonnull)zoom)
RCT_EXTERN_METHOD(resetSelectedId:(nonnull NSNumber *)reactTag)
@end

File: ios/YaMap/YaMapVC.swift
//
//  YaMapVC.swift
//  YaMapTest
//
//  Created by Pavel on 16.08.2023.
//

import Foundation
import React
import YandexMapsMobile
import Combine
//import Apollo

//@available(iOS 15.0, *)
@objc(YaMapVC)
class YaMapVC: RCTViewManager  {
    
  let mapView = YaMapView()
 
  
  
  override init() {
    super.init()
//    let apolloClient = ApolloClient(url: URL(string: "https://apartshering.com/graphql")!)
//    apolloClient.fetch(query: apartments())
  }
   
  @objc func setCenter(_ reactTag: NSNumber, center: Dictionary<NSString, NSNumber>, zoom: NSNumber) {
    self.bridge.uiManager.addUIBlock { manager, dict in
      let view = dict?[reactTag] as? YaMapView
      let centerPoint: Point = Point(lat: center["lat"] as? Double ?? 0, lon: center["lon"] as? Double ?? 0 )
      view?.setCenter(center: centerPoint, zoom: Float(zoom))
    }
    
  }
  
  @objc func resetSelectedId(_ reactTag: NSNumber) {
    self.bridge.uiManager.addUIBlock { manager, dict in
      let view = dict?[reactTag] as? YaMapView
  
      view?.resetSelectedId()
    }

  }
   
   
  override static func requiresMainQueueSetup() -> Bool {
      return true
    }
  
  
  override func view() -> UIView! {
    return mapView
  }
}
